src/decision_model.jl

```diff
@@ -34,141 +45,152 @@ function is_forbidden(s::Path, forbidden_paths::Vector{ForbiddenPath})
-    end
 
     # Create path probability variable for each effective path.
     N = length(S)
-    π_s = Dict{Path{N}, VariableRef}(
-        s => path_probability_variable(model, z, s, P, hard_lower_bound, is_forbidden(s, forbidden_paths), probability_scale_factor, (names ? "$(name)$(s)" : ""))
+    variables_x_s = Dict{Path{N}, VariableRef}(
+        s => path_compatibility_variable(model, z, (names ? "$(name)$(s)" : ""))
         for s in paths(S, fixed)
-        if !iszero(P(s))
+        if !iszero(P(s)) && !is_forbidden(s, forbidden_paths)
     )
-    PathProbabilityVariables{N}(π_s)
-end
 
-"""Adds a probability cut to the model as a lazy constraint.
+    x_s = PathCompatibilityVariables{N}(variables_x_s)
 
-# Examples
-```julia
-probability_cut(model, π_s, P)
-```
-"""
-function probability_cut(model::Model, π_s::PathProbabilityVariables, P::AbstractPathProbability; probability_scale_factor::Float64=1.0)
-    if probability_scale_factor ≤ 0
-        throw(DomainError("The probability_scale_factor must be greater than 0."))
+    # Add decision strategy constraints for each decision node
+    for (d, z_d) in zip(z.D, z.z)
+        decision_strategy_constraint(model, S, d, z.D, z_d, x_s)
     end
 
-    # Add the constraints only once
-    ϵ = minimum(P(s) for s in keys(π_s))
-    flag = false
-    function probability_cut(cb_data)
-        flag && return
-        πsum = sum(callback_value(cb_data, π) for π in values(π_s))
-        if !isapprox(πsum, 1.0 * probability_scale_factor, atol= ϵ * probability_scale_factor)
-            con = @build_constraint(sum(values(π_s)) == 1.0 * probability_scale_factor)
-            MOI.submit(model, MOI.LazyConstraint(cb_data), con)
-            flag = true
-        end
+    if probability_cut
+        @constraint(model, sum(x * P(s) for (s, x) in x_s) == 1.0)
     end
-    MOI.set(model, MOI.LazyConstraintCallback(), probability_cut)
+
+    x_s
 end
 
-"""Adds a active paths cut to the model as a lazy constraint.
+"""
+   lazy_probability_cut(model::Model, x_s::PathCompatibilityVariables, P::AbstractPathProbability)
+
+Adds a probability cut to the model as a lazy constraint.
 
 # Examples
 ```julia
-atol = 0.9  # Tolerance to trigger the creation of the lazy cut
-active_paths_cut(model, π_s, S, P; atol=atol)
+lazy_probability_cut(model, x_s, P)
 ```
-"""
-function active_paths_cut(model::Model, π_s::PathProbabilityVariables, S::States, P::AbstractPathProbability; atol::Float64 = 0.9, probability_scale_factor::Float64=1.0)
-    all_active_states = all(all((!).(iszero.(x))) for x in P.X)
-    if !all_active_states
-        throw(DomainError("Cannot use active paths cut if all states are not active."))
-    end
 
-    if probability_scale_factor ≤ 0
-        throw(DomainError("The probability_scale_factor must be greater than 0."))
-    end
+!!! warning
+Remember to separately set lazy constraints on in the solver parameters (optimizer attributes). For Gurobi `"LazyConstraints" => 1.`
```

Comment:
Gurobi sets the flag automatically whenever you add lazy constraints, so this could be reworded.

Reply:
I understood this differently from this page. https://www.gurobi.com/documentation/9.1/refman/lazyconstraints.html#parameter:LazyConstraints

Where did you find the information that it sets it automatically?

Reply:
Correction: Gurobi.jl sets the flag automatically: https://github.com/jump-dev/Gurobi.jl/blob/6484ba968b09526f13bb8cb5a1fe27abc82bb627/src/MOI_wrapper/MOI_callbacks.jl#L175-L176

Reply:
Okay cool, I'll change the docstring.